src/core/services/compiler/compiler.js

```diff
@@ -10,74 +10,91 @@ angular
 
 /**
  * @ngdoc service
- * @name $mdCompiler
+ * @name $mdCompilerProvider
  * @module material.core.compiler
  * @description
- * The $mdCompiler service is an abstraction of AngularJS's compiler, that allows developers
- * to easily compile an element with options like in a Directive Definition Object.
+ * The `$mdCompiler` is able to respect the AngularJS `$compileProvider.preAssignBindingsEnabled` state when using
+ * AngularJS versions between 1.5.10 and 1.7.0.
+ * See the [AngularJS documentation for `$compile.preAssignBindingsEnabled`
+ * ](https://code.angularjs.org/1.6.10/docs/api/ng/provider/$compileProvider#preAssignBindingsEnabled)
+ * for more information.
  *
- * > The compiler powers a lot of components inside of AngularJS Material.
- * > Like the `$mdPanel` or `$mdDialog`.
+ * To enable/disable whether the controllers of dynamic AngularJS Material components
+ * (i.e. dialog, panel, toast, bottomsheet) respect the AngularJS `$compile.preAssignBindingsEnabled` flag,
+ * call the AngularJS Material method: `$mdCompilerProvider.respectPreAssignBindingsEnabled(boolean)`.
+ *
+ * This AngularJS Material *flag* doesn't affect directives/components created via regular AngularJS methods.
+ * These constitute the majority of AngularJS Material and user-created components.
+ * Only dynamic construction of elements such as Dialogs, Panels, Toasts, BottomSheets, etc. may be affected.
+ * Invoking `$mdCompilerProvider.respectPreAssignBindingsEnabled(true)` will effect **bindings** in
+ * AngularJS Material custom components like `$mdDialog`, `$mdPanel`, `$mdToast`, or `$mdBottomSheet`.
+ *
+ * See [$mdCompilerProvider.respectPreAssignBindingsEnabled](#mdcompilerprovider-respectpreassignbindingsenabled-respected)
+ * for the details of how the different versions and settings of AngularJS affect this behavior.
  *
  * @usage
  *
- * Basic Usage with a template
+ * Respect the AngularJS Compiler Setting
  *
  * <hljs lang="js">
- *   $mdCompiler.compile({
- *     templateUrl: 'modal.html',
- *     controller: 'ModalCtrl',
- *     locals: {
- *       modal: myModalInstance;
- *     }
- *   }).then(function (compileData) {
- *     compileData.element; // Compiled DOM element
- *     compileData.link(myScope); // Instantiate controller and link element to scope.
+ *   app.config(function($mdCompilerProvider) {
+ *     $mdCompilerProvider.respectPreAssignBindingsEnabled(true);
  *   });
  * </hljs>
  *
- * Example with a content element
+ * Assign Bindings Before the Constructor
```

Comment:
This is only true for AngularJS <1.6.0 or AngularJS <1.7.0 if `$compileProvider.preAssignBindingsEnabled(true)` is called. Shouldn't it just say the reverse of the upper example's "Respect the AngularJS Compiler Setting"?